.github/workflows/manual-deploy-dev-staging.yml

```diff
@@ -18,11 +18,98 @@ on:
+        if: ${{ github.event.inputs.deploy_environment == 'dev' || github.event.inputs.deploy_environment == 'both' }}
+        run: yarn build --verbose --buildtype=${{ env.BUILD_ENV_DEV }}
+        timeout-minutes: 30  
+        
+      - name: Generate build details for dev
+        if: ${{ github.event.inputs.deploy_environment == 'dev' || github.event.inputs.deploy_environment == 'both' }}
+        run: |
+          cat > build/${{ env.BUILD_ENV_DEV }}/BUILD.txt << EOF
+          BUILDTYPE=${{ env.BUILD_ENV_DEV }}
+          NODE_ENV=production
+          BRANCH_NAME=$(echo "${GITHUB_REF#refs/heads/}")
+          CHANGE_TARGET=null
+          RUN_ID=${{ github.run_id }}
+          RUN_NUMBER=${{ github.run_number }}
+          REF=${{ github.sha }}
+          BUILDTIME=$(date +%s)
+          EOF
+
+      - name: Compress and archive build for dev 
+        if: ${{ github.event.inputs.deploy_environment == 'dev' || github.event.inputs.deploy_environment == 'both' }}
+        run: tar -C build/${{ env.BUILD_ENV_DEV }} -cjf ${{ env.BUILD_ENV_DEV }}.tar.bz2 .
+
+      - name: Upload build artifact for dev
+        if: ${{ github.event.inputs.deploy_environment == 'dev' || github.event.inputs.deploy_environment == 'both' }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.BUILD_ENV_DEV }}.tar.bz2
+          path: ${{ env.BUILD_ENV_DEV }}.tar.bz2
+          retention-days: 1   
+        
+      - name: Build staging
+        if: ${{ github.event.inputs.deploy_environment == 'staging' || github.event.inputs.deploy_environment == 'both' }}
+        run: yarn build --verbose --buildtype=${{ env.BUILD_ENV_STAGING }}
+        timeout-minutes: 30 
+
+      - name: Generate build details for staging 
+        if: ${{ github.event.inputs.deploy_environment == 'staging' || github.event.inputs.deploy_environment == 'both' }}
+        run: |
+          cat > build/${{ env.BUILD_ENV_STAGING }}/BUILD.txt << EOF
+          BUILDTYPE=${{ env.BUILD_ENV_STAGING }}
+          NODE_ENV=production
+          BRANCH_NAME=$(echo "${GITHUB_REF#refs/heads/}")
+          CHANGE_TARGET=null
+          RUN_ID=${{ github.run_id }}
+          RUN_NUMBER=${{ github.run_number }}
+          REF=${{ github.sha }}
+          BUILDTIME=$(date +%s)
+          EOF
+
+      - name: Compress and archive build for staging
+        if: ${{ github.event.inputs.deploy_environment == 'staging' || github.event.inputs.deploy_environment == 'both' }}
+        run: tar -C build/${{ env.BUILD_ENV_STAGING }} -cjf ${{ env.BUILD_ENV_STAGING }}.tar.bz2 .
+
+      - name: Upload build artifact for staging
+        if: ${{ github.event.inputs.deploy_environment == 'staging' || github.event.inputs.deploy_environment == 'both' }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.BUILD_ENV_STAGING }}.tar.bz2
+          path: ${{ env.BUILD_ENV_STAGING }}.tar.bz2
+          retention-days: 1
+
   deploy:
     name: Deploy
```

Comment:
It would be nice if we could update this job to use a matrix also to remove redundancy, but it's not as impactful as the `Build` job not using a matrix

Reply:
need more testing for matrix type build/deploy. Marking resolve for this sprint considering the priority to be around prod.